payu/models/cesm_cmeps.py

```diff
@@ -163,8 +161,30 @@ def setup(self):
             )
         for realm in all_realms:
             ntasks = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_ntasks"))
-            assert cpucount >= ntasks, "Insufficient cpus for the pelayout in nuopc.runconfig"
-        
+            rootpe = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_rootpe"))
+            pestride = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_pestride"))
+            #rootpe is zero-based
+            assert cpucount >= (rootpe + ntasks*pestride), f"Insufficient cpus for the {realm} pelayout in nuopc.runconfig"
```

Comment:
@jo-basevi - Should I use a runtime error here instead of an asset ??

Reply:
Yes - see https://github.com/payu-org/payu/issues/367

Reply:
Yes, I think there is still a few asserts in payu code in general, but the goal is move towards exceptions instead

Reply:
:) - thanks